### PR TITLE
Networking Coverity fixes

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

File: subsys/net/ip/utils.c [hunk 1]
```diff
@@ -313,7 +313,10 @@ int net_addr_pton(sa_family_t family, const char *src,
 					return -EINVAL;
 				}
 
-				src++;
+				if (src) {
```
Comment (pfalcon) [inline subsys/net/ip/utils.c]: This doesn't look like a code written by human to me. An automated program rewrite tool could do that. A human would refactor to:

~~~
if (src) {
  src++
} else {
  if (i < expected_groups - 1) ...
}
~~~

P.S. Rule of thumb "try to change as little as possible lines/chars in a patch" has its limits too, sometimes it makes sense to do more changes than the absolute minimum necessary, to maintain code quality/clarity.

File: subsys/net/ip/utils.c [hunk 2]
```diff
@@ -367,7 +370,9 @@ int net_addr_pton(sa_family_t family, const char *src,
...
```
Comment (pfalcon) [inline subsys/net/ip/utils.c]: Same comment as above.

File: subsys/net/l2/ethernet/gptp/gptp.c [hunk 3]
```diff
@@ -629,6 +629,10 @@ void gptp_set_time_itv(struct gptp_uscaled_ns *interval,
 			/* Overflow, set maximum. */
 			interval->low = UINT64_MAX;
 			interval->high = UINT32_MAX;
+		} else if (log_msg_interval > 64) {
```
Comment (pfalcon) [inline subsys/net/l2/ethernet/gptp/gptp.c]: Why not add "&&" condition to the "if" above?

**Review by pfalcon — CHANGES_REQUESTED** [review]
In places looks to mechanical a rewrite. Please apply some DRY to the code.

File: subsys/net/ip/utils.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/ip/utils.c]: > } else { if (i < expected_groups - 1) ...

I could not really figure out from where did you got the "else" branch.
With this new code in this commit, we bail out after the continue statement.

File: subsys/net/l2/ethernet/gptp/gptp.c (hunk 3 above)
Comment (jukkar) [reply subsys/net/l2/ethernet/gptp/gptp.c]: That can be done, will send a new version.

File: subsys/net/ip/utils.c (hunk 1 above)
Comment (pfalcon) [reply subsys/net/ip/utils.c]: I got it from looking at a bit more context than just a line Coverity complains about (apparently).

The code you want to land is:

~~~
if (!src && FOO) {
}
if (src) {
}
~~~

Again, humans don't normally write such code. Why instruct compiler to test the same condition twice, of instead if-else can be used?

Comment (jukkar) [reply subsys/net/ip/utils.c]: Ah, got it. Sure, that can be refactored.

**Comment by jukkar** [comment]
New version with utils patch refactored

**Review by pfalcon — APPROVED** [review]
Looks good, thanks for refactoring.

**Pull request merged**